### PR TITLE
Python: model aiomysql

### PR DESCRIPTION
**Pull request by yoff**

### CODE REVIEW

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+
+/** Provides models for the `aiomysql` PyPI package. */
+private module Aiomysql {
+  private import semmle.python.internal.Awaited
+
+  /**
+   * A `ConectionPool` is created when the result of `aiomysql.create_pool()` is awaited.
+   * See https://aiomysql.readthedocs.io/en/stable/core.html#pool
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:19]: This link does not work

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+   */
+  API::Node connectionPool() {
+    result = API::moduleImport("aiomysql").getMember("create_pool").getReturn().getAwaited()
+  }
+
+  /**
+   * A `Connection` is created when
+   * - the result of `aiomysql.connect()` is awaited.
+   * - the result of calling `aquire` on a `ConnectionPool` is awaited.
+   * See https://aiomysql.readthedocs.io/en/stable/core.html#connection
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:29]: neither does this one

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    result = API::moduleImport("aiomysql").getMember("connect").getReturn().getAwaited()
+    or
+    result = connectionPool().getMember("acquire").getReturn().getAwaited()
+  }
+
+  /**
+   * A `Cursor` is created when
+   * - the result of calling `cursor` on a `ConnectionPool` is awaited.
+   * - the result of calling `cursor` on a `Connection` is awaited.
+   * See https://aiomysql.readthedocs.io/en/stable/core.html#cursor
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:41]: or this

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+   */
+  API::Node cursor() {
+    result = connectionPool().getMember("cursor").getReturn().getAwaited()
+    or
+    result = connection().getMember("cursor").getReturn().getAwaited()
+  }
+
+  /**
+   * Calling `execute` on a `Cursor` constructs a query.
+   * See https://aiomysql.readthedocs.io/en/stable/core.html#aiomysql.Cursor.execute
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:51]: or this

File: python/ql/lib/semmle/python/frameworks/Aiomysql.qll [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    exists(DataFlow::TypeTracker t2 | result = cursorExecuteCall(t2, sql).track(t2, t))
+  }
+
+  DataFlow::Node cursorExecuteCall(DataFlow::Node sql) {
+    cursorExecuteCall(DataFlow::TypeTracker::end(), sql).flowsTo(result)
+  }
+
+  /**
+   * Awaiting the result of calling `execute` executes the query.
+   * See https://aiomysql.readthedocs.io/en/stable/core.html#aiomysql.Cursor.execute
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/frameworks/Aiomysql.qll:77]: or this

**Review by RasmusWL — CHANGES_REQUESTED** [review]
A few of the links doesn't work, but otherwise LGTM :+1:

**Comment by yoff** [comment]
Good catch (I really should have checked this), they are probably the result of a search and replace, it is somewhat hilarious that any of them actually worked.. :-)

**Review by RasmusWL — APPROVED** [review]
Thanks, LGTM :+1:

**Pull request merged**